src/track/external-captions-handler.js

```diff
@@ -0,0 +1,345 @@
+import Error from '../error/error'
+import * as Utils from '../utils/util'
+import {Parser, StringDecoder} from './text-track-display'
+import {TrackType} from "./track-type";
+import TextTrack from "./text-track";
+import {CustomEventType, Html5EventType} from "../event/event-type";
+import FakeEvent from "../event/fake-event";
+import getLogger from "../utils/logger";
+
+class ExternalCaptionsHandler {
+
+  static VTT_POSTFIX: string = 'vtt';
+
+  static SRT_POSTFIX: string = 'srt';
+  /**
+   * The player class logger.
+   * @type {any}
+   * @static
+   * @private
+   */
+  static _logger: any = getLogger('External captions handler');
+  /**
+   * Index that specifies the last cue that is playing / played in the text track cue array.
+   * @type {number}
+   * @private
+   */
+  _externalCueIndex: number = 0;
+  /**
+   * a callback of the timeupdate, for external captions handler.
+   * @type {Function}
+   * @private
+   */
+  _handleExternalCaptionsCallback: Function = null;
+  /**
+   * the video element, used for native tracks.
+   * @type {HTMLVideoElement}
+   * @private
+   */
+  _videoElement: HTMLVideoElement = null;
+  /**
+   * the player object.
+   * @type {Object}
+   * @private
+   */
+  _player: null;
+
+  /**
+   * constructor
+   * @param {Object} player - the player object.
+   */
+  constructor(player: Object) {
+    this._player = player;
+    this._videoElement = player.getVideoElement();
+    this._addListeners();
+  }
+
+
+  _addListeners(): void {
+    this._player._eventManager.listen(this._player._engine, this._player.Event.SEEKED, (e)=>{this._maybeSetExternalCueIndex(e)});
```

Comment:
`this._player._engine`
This is access to a private member.
You need to find other solution for this